### ISSUE TITLE
False positive with E1136 for PyQt4 overloaded signal

### ISSUE THREAD

**Issue by alete89** [PyCQA/pylint#2543]
### Steps to reproduce
this code

```python
from PyQt4 import QtGui
...
aComboBox = QtGui.QComboBox(self)
aList = [6,7,8,9,10,11,12,13,14,15,16,18,20,22,24,26,28]
for i in aList:
    aComboBox.addItem(str(i))

fontSizeCombo.activated[str].connect(self.slotOnActivated)
```

where brackets indicate the overloaded signal for strings

### Current behavior
E1136:Value 'aComboBox.activated' is unsubscriptable

There is others false positives reported about this E1136, but neither from this syntax

### Expected behavior
no warning or error.

### pylint --version output
1.9.3

**Comment by PCManticore** [comment]
Thanks for reporting an issue! We'd probably need to add a fix in the corresponding astroid brain transforms: https://github.com/PyCQA/astroid/blob/master/astroid/brain/brain_qt.py#L76

**Comment by Paciupa** [comment]
I have the same issue. With next code:
```python
from PyQt5.QtWidgets import QLineEdit
self.line_edit_variable = QLineEdit(self)
self.line_edit_variable.textChanged[str].connect(self.set_variable)
``` 
I get: `E1136: Value 'self.line_edit_private_key.textChanged' is unsubscriptable (unsubscriptable-object)`.

Versions:
```bash
pylint 2.17.7
astroid 2.15.8
```

**Comment by Pierre-Sassoulas** [comment]
This require a tip to the astroid inference engine, we call that a "brain". As the astroid maintainers are spread very thin it's likely that creating a qt plugin (using the astroid brain's file here: https://github.com/pylint-dev/astroid/blob/main/astroid/brain/brain_qt.py as a basis) is the way to go. We can host it in the pylint-dev namespace so it gathers more attention (like we did for pylint-tensorflow here https://github.com/pylint-dev/pylint-tensorflow).

**Comment by emmanuel-ferdman** [comment]
@alete89 @Paciupa @jaroslavpachola `pylint-qt` plugin was just released and is available on [PyPI](https://pypi.org/project/pylint-qt/)! 🎉
It fixes those Qt false-positives. Please give it a try:
```sh
pip install pylint-qt
pylint --load-plugins=pylint_qt --extension-pkg-allow-list=PyQt5 your_code.py
```
Or in `pyproject.toml`:
```
[tool.pylint.main]
load-plugins = ["pylint_qt"]
extension-pkg-allow-list = ["PyQt5"]
```
repo: https://github.com/pylint-dev/pylint-qt

**Comment by Pierre-Sassoulas** [comment]
Thank you for your work on this @emmanuel-ferdman, let's hope a lot of astroid plugins spawn from your example 💪
